bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingImpl.java

```diff
@@ -201,6 +208,33 @@ public boolean isLinked() {
     }
 
     @Override
+    public Map<String, String> getProperties() {
+        synchronized (this) {
+            return ImmutableMap.copyOf(properties);
+        }
+    }
+
+    @Override
+    public void setProperty(String name, String value) {
```

Comment:
Why is the return type Void? Map.put(k,v) returns the old value, which is quite useful to check for real changes.